### ISSUE TITLE
Flickering when placing windows

### ISSUE THREAD

**Issue by kaihendry** [koekeishiya/kwm#544]
koekeishiya/kwm/kwm: stable 4.0.2

I don't have a configuration at all on a new MBP running 10.12.2 (16C67).

But I noticed an awful "flickering" when it's trying to place a window at times.

Also some windows like "About this mac" don't get tiled and disappear behind by browser.

Many thanks and merry Christmas,

**Comment by koekeishiya** [comment]
> I don't have a configuration at all on a new MBP running 10.12.2 (16C67).

This may sound kinda silly, but I am actually not sure of the expected behaviour with no config at all.
Copying the example config and tweaking from there is highly recommended.

> But I noticed an awful "flickering" when it's trying to place a window at times.

I assume this only happens for some windows / applications? If you have some more info regarding this, that would be great.

> Also some windows like "About this mac" don't get tiled and disappear behind by browser.

These are basically windows that Kwm doesn't recognize due to inconsistency in the OSX Accessibility APIs, regarding window roles. I can get more in depth here if you want, but basically a lot of roles are used for menu-items, context-menus, tooltips and so on, and these are all filtered out by Kwm. Windows such as "About this mac" happens to fail the filter check.

I guess technically I could have a blacklist for certain OSX specific windows that we know are actual windows though.

**Comment by chrishoage** [comment]
> I assume this only happens for some windows / applications? If you have some more info regarding this, that would be great.

I have more info, this seems to happen with windows that "suggest" a size. For example. iTerm can "flicker" when being tiled for a few seconds until it finally settles.

It seems as if iTerm wants to be a very specific size, and kwm keeps trying to resize it. They both eventually figure it out. 

This does not _always_ happen, but I have seen it happen before. Unfortunately I don't have 100% reproducible steps.

**Comment by dominiklohmann** [comment]
> It seems as if iTerm wants to be a very specific size, and kwm keeps trying to resize it. 

Regardless of the issue (I can reproduce it on Sierra), you can disable that behaviour from iTerm. For the time being this seems to be a fix for iTerm at least. 

The cause seems to be a resize loop (Kwm resizes -> iTerm snaps to grid -> Kwm resizes -> ...), and after some time Kwm seems to give up(?).

<img width="1130" alt="screen shot 2017-01-10 at 19 26 25" src="https://cloud.githubusercontent.com/assets/4488655/21819132/b71edade-d76a-11e6-80ba-9937eea60064.png">

**Comment by choco** [comment]
Yeah this is usually an issue with windows smaller than the node container that aren't resizable or have some specific constraint. We probably need to implement a better heuristic in `CenterWindowInsideNodeContainer`

**Comment by pablocobelli** [comment]
@dominiklohmann Thanks for the iTerm2 configuration tip. I confirmed I observed the 'flickering' behaviour described by @chrishoage; @dominiklohmann 's tip solved it for me.
